### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by alexAubin**
## Changes

New permission system

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by alexAubin** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Review by ericgaspar — APPROVED** [review]

File: scripts/install [hunk 1]
```diff
@@ -141,21 +148,29 @@ chown -R $app: "$final_path/"{tpl_c,admin/stdout.log}
...
 	    # If the path is /, clear it to prevent any error with the regex.
 	    path_url=""
 	fi
 	# Modify the domain to be used in a regex
 	domain_regex=$(echo "$domain" | sed 's@-@.@g')
-	ynh_app_setting_set --app=$app --key=protected_regex --value="$domain_regex$path_url/admin/"
+	ynh_app_setting_set --app=$app --key=unprotected_regex --value="$domain_regex$path_url/create_poll.php?.*$","$domain_regex$path_url/adminstuds.php?.*"
+else
+	ynh_permission_update --permission="create poll" --remove="visitors"
 fi
```
Comment (alexAubin) [inline scripts/install:174]: Hmpf in fact there are several issues that would be nice to address ...
- the id of the permission should not contain spaces
- this is still relying on the legacy permission system and we really need to find a way to use the new permission system instead ... The comment states that the core doesn't handle regexes but I think nowaday the new permission does (? not sure ..)
- The current regex thing explodes for domains containing a `-`, c.f. for example https://forum.yunohost.org/t/probleme-dinstallation-avec-la-creation-des-permissions/17816
- I don't understand the need to differentiate `if is_public` / `else` that act on two different permissions x_x Apparently that's related to the regex thing not being supported and the hook system but it looks like it's gonna create huge inconsistencies with respect to what the admin may do in the webadmin in the permission screen later ? I mean if the app is installed as private, the unprotected_regex is never created at all ... x_X

Comment (alexAubin) [reply scripts/install:174]: One way to solve the issue is to say "fuck it" and simply have a classic public/private where if the app is public, anybody can view and create polls, and if private, only registered users can view and create polls ... I don't think it's a huge deal to not have a specific permission for poll creation ...

**Comment by alexAubin** [comment]
!testme

**Pull request merged**